uk.co.compucorp.civicrm.tasksassignments/api/v3/Document.php

```diff
@@ -34,8 +34,17 @@ function civicrm_api3_document_create($params) {
     return $errors;
   }
 
+  // Processing for custom data
+  $customFields = _civicrm_api3_document_getcustomfieldsflipped();
+  foreach ($customFields as $key => $value) {
+    if (isset($params[$key])) {
+      $params[$value] = $params[$key];
```

Comment:
are you sure that $params[$value]  is correct assignment, because if I am following the code correctly $customFields array should be in this format : 

array (
'custom_field_name_1' => 0,
'custom_field_name_2' => 1,
'custom_field_name_3' => 2,
..etc
)

so $params[0] = $params['custom_field_name_1'] doesn't seems like something make sense since $params[0] represents nothing here.